### ISSUE TITLE
Rollup warning: `this` has been rewritten to `undefined`

### ISSUE THREAD

**Issue by pablo-abc** [github/template-parts#52]
Hello! I've recently been experimenting with different alternatives to use web components and I found catalyst, which I've loved! I tried adding `@github/jtml` to a component I'm making and I'm getting the following warning when bundling with Rollup:

```
(!) `this` has been rewritten to `undefined`
https://rollupjs.org/guide/en/#error-this-is-undefined
../../node_modules/.pnpm/@github+template-parts@0.3.3/node_modules/@github/template-parts/lib/attribute-template-part.js
1: var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
                                 ^
2:     if (kind === "m") throw new TypeError("Private method is not writable");
3:     if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
...and 3 other occurrences
../../node_modules/.pnpm/@github+template-parts@0.3.3/node_modules/@github/template-parts/lib/template-instance.js
1: var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
                                 ^
2:     if (kind === "m") throw new TypeError("Private method is not writable");
3:     if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
...and 3 other occurrences
../../node_modules/.pnpm/@github+template-parts@0.3.3/node_modules/@github/template-parts/lib/node-template-part.js
1: var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
                                 ^
2:     if (kind === "m") throw new TypeError("Private method is not writable");
3:     if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
...and 3 other occurrences
```

From reading [the link on the message](https://rollupjs.org/guide/en/#error-this-is-undefined) and seeing the error message contents, this seems to be intentional on your side, so this might just need to be documented? Setting [`context`](https://rollupjs.org/guide/en/#context) to `window` removes the warning and doesn't seem to break anything, but I'm not sure if this would be the proper approach.

**Comment by keithamus** [comment]
This is a great catch @pablo-abc! Would you kindly set up a PR documenting the caveat you've seen here?

**Comment by pablo-abc** [comment]
Can do that @keithamus! Just a question. So the issue seems to come from the fact that there's no global `this` on ES modules. But tsc compiles to code that checks if the helper `__classPrivateFieldSet` is on the global `this` before assigning its own function. [This is an intentional behaviour](https://github.com/microsoft/TypeScript/issues/43728) since tsc assumes the code _might_ not run on an ES module, but seeing as this package is meant to always be used as an ES module, this does not make sense due to `this` _always_ being undefined in this case.

I found when that using `"importHelpers": true` and `tslib` this never happens since tsc just imports the helper from `tslib` without checking for a global `this`. _But_ this would require bundling tslib (using a bundler) instead of plain `tsc`. I always use Rollup for my TS projects so I've never faced this issue before. I'm not sure if there'd be a way to prevent tsc from generating that code or just adding a node to the README would be the best approach. [Other libraries did switch to Rollup from tsc to solve this](https://github.com/statelyai/xstate/pull/1597). I tried doing this as well and it does indeed work as expected.

Let me know what you think!
